Engineering/Frontend/WAI-ARIA.md

```diff
@@ -0,0 +1,70 @@
+---
+tags: engineering/frontend, wai-aria, accessibility
+author: Nguyen Dinh Nam
+date: 2022-09-30
+---
+
+Since modern websites and applications contain huge amounts of dynamic content and complex components with no semantics to describe what they mean, users with **no mouse** equiped or those dependent on technologies such as **[Screen Reader](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn/related?hl=en)** might not be able to interact with those components.
+
+Web Accessibility Initiative’s Accessible Rich Internet Applications or simply **WAI-ARIA** is a technology that can help with such problems.
+
+## What is WAI-ARIA?
+Before semantic elements like `<nav>` or `<footer>` were introduced that define specific features of a web page, some developers would rely on JavaScript libraries that generate a bunch of **nested `<div>`s**, then styled them with CSS and controlled with JavaScript. 
+
+Even though they work fine, screen readers cannot make any sense of what those components are, no semantics provided so that they can assist their users. WAI-ARIA has **roles**, **states** and **properties** to solve this problem, giving an opportunity to add attributes to content and components that make them **meaningful** and enhance **accessibility**.
```

Comment:
You might want to add a small list here, listing out the next 3 things that you are going to introduce to the readers. E.g.:

```
3 main features of WAI-ARIA:
- Roles
- Properties
- States
```

You can add some short description (or ref link) to each item. This is a way to summarize & let the readers know in advance what they should expect to see next.